src/sql/base/parts/editableDropdown/browser/dropdown.ts

```diff
@@ -263,9 +263,11 @@ export class Dropdown extends Disposable implements IListVirtualDelegate<string>
 		this._selectListContainer.setAttribute('aria-hidden', `${!visible}`);
 	}
 
-	private _updateSelection(newValue: string): void {
-		this.value = newValue;
-		this._onValueChange.fire(newValue);
+	public updateSelection(newValue: string): void {
+		if (this.value !== newValue) {
+			this.value = newValue;
+			this._onValueChange.fire(newValue);
```

Comment:
I would think that setting value should fire the event. Why would we only fire this when updateSelection is called specifically?